src/main/java/seedu/mark/model/reminder/ReminderAssociation.java

```diff
@@ -0,0 +1,119 @@
+package seedu.mark.model.reminder;
+
+import static seedu.mark.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Comparator;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.collections.ObservableMap;
+
+import seedu.mark.model.bookmark.Bookmark;
+import seedu.mark.model.bookmark.exceptions.BookmarkContainNoReminderException;
+import seedu.mark.model.bookmark.exceptions.ExistReminderException;
+import seedu.mark.model.bookmark.exceptions.ReminderNotFoundException;
+
+/**
+ * Represents the association between bookmarks and reminders.
+ */
+public class ReminderAssociation {
+    private static final Comparator<Reminder> comparator = (Reminder a, Reminder b) -> {
+        return a.getRemindTime().isBefore(b.getRemindTime()) ? -1 : 1;
+    };
+    //TODO: One bookmark may has multiple reminder in next version.
+    private ObservableMap<Bookmark, Reminder> association = FXCollections.observableHashMap();
+
+    /**
+     * Sets the reminder association with the given association.
+     *
+     * @param association the given association.
+     */
+    public void setAssociation(ObservableMap<Bookmark, Reminder> association) {
+        this.association = association;
+    }
+
+    /**
+     * Adds a reminder that opens the bookmark.
+     *
+     * @param bookmark the bookmark that is opened by the reminder.
```

Comment:
```suggestion
     * @param bookmark The bookmark that the reminder opens.
```

Reply:
I personally feel that capitalising the first word after the param name makes it easier to read; we can consider which style is nicer to follow

Reply:
Okay